crates/starlight/src/vm/value.rs

```diff
@@ -681,7 +681,10 @@ impl JsValue {
             println!("{:?}", (object.get_dyn()).type_name());
             todo!()
         } else {
-            unreachable!("{:?}", self.type_of())
+            Err(JsValue::new(ctx.new_type_error(format!(
```

Comment:
This one should be unreachable instead of throwing an error. If this case is reached then some UB happened.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toString
Here, the symbol to_string concatenation return TypeError ?

Reply:
Hmm that's true. It should be handled in different way. I will add special function `to_string_for_concat` later that will throw error on symbol but in current impl `to_string` already checks for `Symbol` object in `is_object` branch.